### PR TITLE
Add cluster login cmd

### PR DESCRIPTION
**Pull request by wanghaoran1988**
This add cmd : uhc cluster login <cluster_id> to oc login to openshift cluster.
Should be helpful when you try to access the cluster by id.

### CODE REVIEW

**Comment by wanghaoran1988** [comment]
This error didn't happen on my local env:
```
cmd/uhc/cluster/login/login.go:128: G204: Subprocess launched with variable (gosec)
	if err := syscall.Exec(path, append([]string{path, "login"}, url), os.Environ()); err != nil {
make: *** [lint] Error 1
```

My golint version:
golangci-lint --version                                                                                                                                           
golangci-lint has version 1.15.0 built from 901cf25 on 2019-02-18T08:24:43Z

File: cmd/uhc/cluster/login/login.go
Comment (cblecker) [inline cmd/uhc/cluster/login/login.go]: Is there a reason you're using `syscall.Exec` over `os.Exec`? Calling `syscall` directly is unsafe.

Something like should work without calling syscall directly:
```go
import os/exec
cmd := exec.Command("oc", "login", url)
err := cmd.Run()
if err != nil {
  return "", err
}
```

**Comment by aweiteka** [comment]
cc @ALimaRedHat

**Comment by aweiteka** [comment]
+1 to the use case. I was considering the same thing.

File: cmd/uhc/cluster/login/login.go
Comment (jhernand) [inline cmd/uhc/cluster/login/login.go]: This comment ^ isn't accurate, you are retrieving one individual cluster, not the collection.

File: cmd/uhc/cluster/login/login.go
Comment (jhernand) [inline cmd/uhc/cluster/login/login.go]: I try to keep a style of comments consistent across the project: they start with uppercase, end with punctuation and have a blank like before. So in this case it would be:

```go
...
}

// Get cluster API for login:
url = cluster.API().URL()
...
```

Can you use the same style?

File: cmd/uhc/cluster/login/login.go
Comment (jhernand) [inline cmd/uhc/cluster/login/login.go]: Messages shown to the user should start with uppercase and end with new line, so this should be:

```
fmt.Fprintf(os.Stderr(os.Stderr, "Can't ... cluster: %s\n", argv[0])
```

Also you will probably want to `os.Exit(1)` when this happens.

File: cmd/uhc/cluster/login/login.go
Comment (jhernand) [reply cmd/uhc/cluster/login/login.go]: I agree it is better to use `os.Exec`, please do it. However _golangci-lint_ will still complain. To avoid that add a comment like this before the `os.Exec` line:

```
// #nosec G204
```

**Review by jhernand — CHANGES_REQUESTED** [review]
Very nice @wanghaoran1988, thanks! Please address the review comments and I will merge.

**Comment by jhernand** [comment]
@wanghaoran1988 another thing to consider, but not for this patch, is that if you have the right permissions the API will give you access to the _kubeadmin_ password and to the `kubeconfig` file of the cluster. Using that, when available, it would be possible to login to the cluster without having to type any user name or password.

**Comment by aweiteka** [comment]
> if you have the right permissions the API will give you access to the _kubeadmin_ password and to the `kubeconfig` file of the cluster. Using that, when available, it would be possible to login to the cluster without having to type any user name or password.

We want to discourage logging in via kubeadmin user. In fact, most SREs will not have access to it due to compliance requirements. Kubeadmin will be reserved for a "break glass" scenario.

**Comment by jhernand** [comment]
> We want to discourage logging in via kubeadmin user. In fact, most SREs will not have access to it due to compliance requirements. Kubeadmin will be reserved for a "break glass" scenario.

I understand, and agree.

File: cmd/uhc/cluster/login/login.go
Comment (fahlmant) [inline cmd/uhc/cluster/login/login.go:149]: `oc login $URL` prompts you for a username and password. I haven't tested your code directly, but does this allow for interactive input?

Comment (wanghaoran1988) [reply cmd/uhc/cluster/login/login.go:149]: @fahlmant Yes, that's how the line 132,133 works.

**Comment by wanghaoran1988** [comment]
@jhernand Comments addressed, could you take a look?

**Comment by cblecker** [comment]
Tested this locally, and it works as expected :)

File: cmd/uhc/cluster/login/login.go
Comment (jhernand) [inline cmd/uhc/cluster/login/login.go]: Pleae put each argument to these methods in a separate line, as we do in all the other commands:

```go
flags.BoolBar(
    &args.debug,
    "debug",
    false,
    "Enable debug mode.",
)
```

File: cmd/uhc/cluster/login/login.go
Comment (jhernand) [inline cmd/uhc/cluster/login/login.go]: Passing the password in the command line is a bad practice. You will probably be asked to remove it in an hypothetical security review. I have been asked to do that in the past, in other projects. Can you remove it?

File: cmd/uhc/cluster/login/login.go
Comment (jhernand) [inline cmd/uhc/cluster/login/login.go:58]: Add a blank line here.

File: cmd/uhc/cluster/login/login.go
Comment (jhernand) [inline cmd/uhc/cluster/login/login.go:69]: Add a blank line here.

File: cmd/uhc/cluster/login/login.go
Comment (jhernand) [inline cmd/uhc/cluster/login/login.go]: This comment isn't accurate.

File: cmd/uhc/cluster/login/login.go
Comment (jhernand) [inline cmd/uhc/cluster/login/login.go]: Remove this blank line ...

File: cmd/uhc/cluster/login/login.go
Comment (jhernand) [inline cmd/uhc/cluster/login/login.go:130]: ... and add it here.

File: cmd/uhc/cluster/login/login.go
Comment (wanghaoran1988) [reply cmd/uhc/cluster/login/login.go]: done

File: cmd/uhc/cluster/login/login.go
Comment (jhernand) [inline cmd/uhc/cluster/login/login.go]: I think that `args != ""` is clearer, consider using that instead of `len(args.user) != 0`.

File: cmd/uhc/cluster/login/login.go
Comment (jhernand) [inline cmd/uhc/cluster/login/login.go:151]: Add `os.Exit(1)`.

**Review by jhernand — COMMENTED** [review]
Looks good @wanghaoran1988. My only relevant comment is that I think we should remove the password option, as passing passwords in the command line isn't safe, in general.

I have also other minor requests.

File: cmd/uhc/cluster/login/login.go
Comment (wanghaoran1988) [reply cmd/uhc/cluster/login/login.go]: sure

**Comment by wanghaoran1988** [comment]
@jhernand Comments addressed.

**Review by jhernand — APPROVED** [review]

**Comment by jhernand** [comment]
Looks good, thanks @wanghaoran1988. Before merging can you please rebase and squash to one commit?

**Comment by jhernand** [comment]
[test]

**Comment by jmelis** [comment]
[test]

**Comment by jhernand** [comment]
@wanghaoran1988 now that tests pass I will merge as soon as you squash to one commit. Then I will do a new release.

**Comment by wanghaoran1988** [comment]
@jhernand squashed

**Pull request merged**
